### PR TITLE
Improve flytectl config init UX 

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- In `config init`, Added logic for timing host name (Now host will trim `dns://`, `https://` & `http://` )
- Now in `config init` default value of insecure would be false

Testing 
- flytectl config init --host=google (Will fail)
- flytectl config init  --host=127.0.0.1 (will pass)
- flytectl config init  --host=localhost (will pass)
- flytectl config init  --host=127.0.0.1.0 (will fail)

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [x] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1880

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: cmd/configuration/configuration.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ var prompt = promptui.Select{
 	Items: []string{"S3", "GCS"},
 }
 
+var dnsPrefix = [3]string{"dns://", "http://", "https://"}
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:54]: dnsPrefix -> endpointPrefixes

File: cmd/config/subcommand/config/init_flags.go [hunk 2]
```diff
@@ -3,7 +3,7 @@ package config
 //go:generate pflags Config --default-var DefaultConfig --bind-default-var
 var (
 	DefaultConfig = &Config{
-		Insecure: true,
+		Insecure: false,
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/config/init_flags.go:6]: Can we also document this in flytectl docs where ever these options are referenced

File: cmd/configuration/configuration.go [hunk 3]
```diff
@@ -114,3 +117,11 @@ func initFlytectlConfig(ctx context.Context, reader io.Reader) error {
...
 	return nil
 }
+
+func trim(hostname string) string {
+	for _, prefix := range dnsPrefix {
+		hostname = strings.Trim(hostname, prefix)
+	}
+	return hostname
+
+}
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:127]: Lets add unit tests for this function

File: cmd/configuration/configuration.go [hunk 4]
```diff
@@ -77,12 +79,13 @@ func initFlytectlConfig(ctx context.Context, reader io.Reader) error {
...
 	templateValues := configutil.ConfigTemplateSpec{
 		Host:     "dns:///localhost:30081",
-		Insecure: initConfig.DefaultConfig.Insecure,
+		Insecure: true,
 	}
 	templateStr := configutil.GetSandboxTemplate()
 
 	if len(initConfig.DefaultConfig.Host) > 0 {
-		templateValues.Host = fmt.Sprintf("dns:///%v", initConfig.DefaultConfig.Host)
+		templateValues.Host = fmt.Sprintf("dns://%s", trim(initConfig.DefaultConfig.Host))
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:87]: After setting this can we validate that its valid dns endpoint

File: cmd/config/subcommand/config/init_flags.go (hunk 2 above)
Comment (pmahindrakar-oss) [reply cmd/config/subcommand/config/init_flags.go:6]: Also does this cause any changes to the sandbox config.

Comment (evalsocket) [reply cmd/config/subcommand/config/init_flags.go:6]: no

File: cmd/configuration/configuration.go (hunk 4 above)
Comment (evalsocket) [reply cmd/configuration/configuration.go:87]: we don't need that extra check

**Review by pmahindrakar-oss — DISMISSED** [review]

File: cmd/configuration/configuration.go (hunk 4 above)
Comment (pmahindrakar-oss) [reply cmd/configuration/configuration.go:93]: It would be useful if user enters garbage data eg : https://0.0.0.0.0.0
Url.Parser might be able to catch it i suppose.

File: cmd/configuration/configuration_test.go [hunk 5]
```diff
@@ -56,8 +56,22 @@ func TestSetupConfigFunc(t *testing.T) {
 	assert.Nil(t, initFlytectlConfig(ctx, yes))
 	assert.Nil(t, initFlytectlConfig(ctx, yes))
 	assert.Nil(t, initFlytectlConfig(ctx, no))
-	initConfig.DefaultConfig.Host = "test"
+	initConfig.DefaultConfig.Host = "flyte.org"
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration_test.go:59]: Can we also add a test with localhost:30081 ie one with host:port

**Review by pmahindrakar-oss — APPROVED** [review]
We shouldn't allow updates like these too. 
```
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns://localhost:30081:1
```
Not a high priority though. We should see if there is already a dns endpoint validation library in apimachinery
